internal/pkg/override/cdk.go

```diff
@@ -16,30 +16,70 @@ import (
 	"github.com/spf13/afero"
 )
 
-// Executable is the interface to wrap os/exec calls.
-type Executable interface {
-	LookPath(file string) (string, error)
-	Command(name string, args ...string) *exec.Cmd // TODO(efe): Command should set cmd.Dir = rootAbsPath.
-}
-
 // CDK is an Overrider that can transform a CloudFormation template with the Cloud Development Kit.
 type CDK struct {
 	rootAbsPath string // Absolute path to the overrides/ directory.
 
-	out  io.Writer  // Writer for any os/exec calls.
-	fs   afero.Fs   // OS file system.
-	exec Executable // For testing os/exec calls.
+	out  io.Writer // Writer for any os/exec calls.
+	fs   afero.Fs  // OS file system.
+	exec struct {
+		LookPath func(file string) (string, error)
+		Command  func(name string, args ...string) *exec.Cmd
```

Comment:
Will this be a problem for mock for unit test?

Reply:
Nope! In the unit tests we use the public constructor to define our own custom behavior:
```
WithCDK("some/path", CDKOpts{
   LookPathFn: func(string) (string, error) { return "", errors.New("some error") }
})
```